kermit-io/src/commonMain/kotlin/co/touchlab/kermit/io/RollingFileLogWriter.kt

```diff
@@ -136,8 +134,8 @@ open class RollingFileLogWriter(
   private suspend fun writer() {
     val logFilePath = pathForLogIndex(0)
 
-    if (fileSystem.exists(logFilePath)) {
-      maybeRollLogs(fileSizeOrZero(logFilePath))
+    if (fileSystem.exists(logFilePath) && shouldRollLogs(logFilePath)) {
```

Comment:
AFAICT we should just do this:
```suggestion
    if (shouldRollLogs(logFilePath)) {
```

because `metadataOrNull()` already handled the case where the file doesn't exist.